File: cpp/behavior/strategy.cpp
#include <iostream>
using namespace std;
class IStategy {
public:
    virtual void call() = 0;
};
class StrategyA: public IStategy {
public:
    void call() {
        cout<<"StrategyA call" <<endl;
    }
};
class StrategyB: public IStategy {
public:
    void call() {
        cout<<"StrategyB call" <<endl;
    }
};

int main() {
    IStategy* strategy = new StrategyA();
    strategy->call();
    return 0;
}

File: golang/behavior/strategy.go
package main

type Strategy interface {
	call()
}
type StrategyA struct {
}
func (sa *StrategyA) call() {
	println("sa call")
}
type StrategyB struct {
}
func (sb *StrategyB) call() {
	println("sb call")
}

var Factory map[int]Strategy = make(map[int]Strategy)
func init() {
	Factory[0] = &StrategyA{}
	Factory[1] = &StrategyB{}
}
func main() {
	var sa Strategy = Factory[1]
	sa.call()
}